pysat/tests/test_instruments.py

```diff
@@ -22,56 +29,187 @@
 # Developers for instrument libraries should update the following line to
 # point to their own subpackage location
 # e.g.,
-# instruments = generate_instrument_list(inst_loc=mypackage.inst)
+# InstLibTests.initialize_test_package(InstLibTests, inst_loc=mypackage.inst)
+
 # If user and password info supplied, use the following instead
-# instruments = generate_instrument_list(inst_loc=mypackage.inst,
-#                                        user_info=user_info)
-instruments = generate_instrument_list(inst_loc=pysat.instruments,
-                                       user_info=user_info)
-
-# The following lines apply the custom instrument lists to each type of test
-method_list = [func for func in dir(InstTestClass)
-               if callable(getattr(InstTestClass, func))]
-# Search tests for iteration via pytestmark, update instrument list
-for method in method_list:
-    if hasattr(getattr(InstTestClass, method), 'pytestmark'):
-        # Get list of names of pytestmarks
-        Nargs = len(getattr(InstTestClass, method).pytestmark)
-        names = [getattr(InstTestClass, method).pytestmark[j].name
-                 for j in range(0, Nargs)]
-        # Add instruments from your library
-        if 'all_inst' in names:
-            mark = pytest.mark.parametrize("inst_name", instruments['names'])
-            getattr(InstTestClass, method).pytestmark.append(mark)
-        elif 'download' in names:
-            mark = pytest.mark.parametrize("inst_dict", instruments['download'])
-            getattr(InstTestClass, method).pytestmark.append(mark)
-        elif 'no_download' in names:
-            mark = pytest.mark.parametrize("inst_dict",
-                                           instruments['no_download'])
-            getattr(InstTestClass, method).pytestmark.append(mark)
-
-
-class TestInstruments(InstTestClass):
-    """Uses class level setup and teardown so that all tests use the same
-    temporary directory. We do not want to geneate a new tempdir for each test,
-    as the load tests need to be the same as the download tests.
+# InstLibTests.initialize_test_package(InstLibTests, inst_loc=mypackage.inst,
+#                                       user_info=user_info)
+
+# If custom tests need to be added to the class, the instrument lists may be
+# included as an optional output.
+# instruments = InstLibTests.initialize_test_package(InstLibTests,
+#                                                    inst_loc=mypackage.inst)
+instruments = InstLibTests.initialize_test_package(InstLibTests,
+                                                   inst_loc=pysat.instruments,
+                                                   user_info=user_info)
```

Comment:
Do we need the comment that gives an example when we have the actual command right here?